tests/prov/test_prov_k8s_cortx_upgrade.py

```diff
@@ -0,0 +1,185 @@
+from commons import constants as cons
+from commons.helpers.pods_helper import LogicalNode
+from commons.utils import system_utils, assert_utils, ext_lbconfig_utils
+from commons.utils import support_bundle_utils
+from commons.params import LOG_DIR
+from commons.params import LATEST_LOG_FOLDER
+from config import CMN_CFG, PROV_CFG, PROV_TEST_CFG
+from libs.ha.ha_common_libs_k8s import HAK8s
+from libs.prov.prov_k8s_cortx_deploy import ProvDeployK8sCortxLib
+
+LOGGER = logging.getLogger(__name__)
+
+
+class TestProvK8CortxUpgrade:
+    """
+    This class contains Provisioner Component level test cases for K8s CORTX Software Upgrade.
+    """
+
+    @classmethod
+    def setup_class(cls):
+        """Setup class"""
+        LOGGER.info("STARTED: Setup Module operations")
+        cls.que = queue.Queue()
+        cls.repo_clone_path = "root"
+        cls.deployment_version = os.getenv("DEPLOYMENT_VERSION")
+        cls.upgrade_image = os.getenv("UPGRADE_IMAGE",None)
+        cls.current_image = os.getenv("CURRENT_IMAGE", None)
+        cls.deploy_conf = PROV_CFG["k8s_cortx_deploy"]
+        cls.prov_deploy_cfg = PROV_TEST_CFG["k8s_prov_cortx_deploy"]
+        cls.deploy_lc_obj = ProvDeployK8sCortxLib()
+        cls.num_nodes = len(CMN_CFG["nodes"])
+        cls.worker_node_list = []
+        cls.master_node_list = []
+        cls.host_list = []
+        cls.remote_path = cons.CLUSTER_CONF_PATH
+        cls.local_path = cons.LOCAL_CONF_PATH
+        cls.collect_sb = True
+        for node in range(cls.num_nodes):
+            node_obj = LogicalNode(hostname=CMN_CFG["nodes"][node]["hostname"],
+                                   username=CMN_CFG["nodes"][node]["username"],
+                                   password=CMN_CFG["nodes"][node]["password"])
+            cls.host_list.append(node_obj)
+            if CMN_CFG["nodes"][node]["node_type"].lower() == "master":
+                cls.master_node_obj = node_obj
+                cls.master_node_list.append(node_obj)
+            else:
+                cls.worker_node_list.append(node_obj)
+        resp = cls.deploy_lc_obj.check_s3_status(cls.master_node_obj, pod_prefix=cons.POD_NAME_PREFIX)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Done: Setup operations finished.")
+
+    @classmethod
+    def teardown_class(self):
+        """"
+        Teardown method
+        """
+        if self.collect_sb:
+            path = os.path.join(LOG_DIR, LATEST_LOG_FOLDER)
+            support_bundle_utils.collect_support_bundle_k8s(local_dir_path=path,
+                                                            scripts_path=
+                                                            self.deploy_conf['k8s_dir'])
+        resp = self.deploy_lc_obj.destroy_setup(self.master_node_list[0],
+                                                self.worker_node_list)
+        assert_utils.assert_true(resp)
+
+    def perform_upgrade(self, upgrade_image_version: str, exc: bool = True):
```

Comment:
This method is not required, service_upgrade_software can be used directly in test.